mobile_verifier/src/server.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{env_var, error::Result, subnetwork_rewards::SubnetworkRewards};
+use chrono::{DateTime, Duration, NaiveDateTime, Utc};
+use helium_proto::services::{follower, Channel, Endpoint, Uri};
+use poc_store::FileStore;
+use tokio::{select, time::sleep};
+
+pub struct Server {
+    pub input_store: FileStore,
+    pub output_store: FileStore,
+    pub follower_client: follower::Client<Channel>,
+    pub last_reward_end_time: i64,
+    pub lookup_delay: i64,
+}
+
+pub const CONNECT_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(5);
+pub const RPC_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(5);
+pub const DEFAULT_URI: &str = "http://127.0.0.1:8080";
+
+/// Default hours to delay lookup from now
+pub const DEFAULT_LOOKUP_DELAY: i64 = 3;
+
+impl Server {
+    pub async fn new() -> Result<Self> {
+        Ok(Self {
+            input_store: FileStore::from_env_with_prefix("INPUT").await?,
+            output_store: FileStore::from_env_with_prefix("OUTPUT").await?,
+            follower_client: follower::Client::new(
+                Endpoint::from(env_var("FOLLOWER_URI", Uri::from_static(DEFAULT_URI))?)
+                    .connect_timeout(CONNECT_TIMEOUT)
+                    .timeout(RPC_TIMEOUT)
+                    .connect_lazy(),
+            ),
+            last_reward_end_time: env_var("LAST_REWARD_END_TIME", 0)?,
+            lookup_delay: env_var("LOOKUP_DELAY", DEFAULT_LOOKUP_DELAY)?,
+        })
+    }
+
+    pub async fn run(self, shutdown: triggered::Listener) -> Result {
+        tracing::info!("Starting verifier service");
+
+        let Self {
+            input_store,
+            output_store,
+            follower_client,
+            last_reward_end_time,
+            lookup_delay,
+        } = self;
+
+        let mut last_reward_end_time =
+            DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(last_reward_end_time, 0), Utc);
+        loop {
+            let (start, stop) = get_time_range(last_reward_end_time, lookup_delay);
+            SubnetworkRewards::from_period(&input_store, follower_client.clone(), start, stop)
+                .await?
+                .write(&output_store)
+                .await?;
+            last_reward_end_time = stop;
```

Comment:
this doesn't appear to be persisting anywhere, does that mean that if we want to preserve the time the server left off processing we need to track it outside of the server somewhere and inject it into the restart by way of the `LAST_REWARD_END_TIME` environment variable or else it'll restart processing from 1, 1, 1970?

Reply:
Yes. I can add this value to the Meta db but I wasn't sure what the preferred way to track this is